map_test.go

```diff
@@ -99,45 +99,61 @@ func TestMapBatch(t *testing.T) {
 	contents := map[uint32]uint32{
 		0: 42, 1: 4242, 2: 23, 3: 2323,
 	}
-
-	hash, err := NewMap(&MapSpec{
-		Type:      Hash,
-		KeySize:   4,
-		ValueSize: 4,
-		// Make the map large enough to avoid ENOSPC.
-		MaxEntries: uint32(len(contents)) * 10,
-	})
-	if err != nil {
-		t.Fatal(err)
+	mustNewMap := func(mapType MapType, max uint32) *Map {
+		m, err := NewMap(&MapSpec{
+			Type:       mapType,
+			KeySize:    4,
+			ValueSize:  4,
+			MaxEntries: max,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		return m
 	}
+
+	var (
+		/* Make the map large enough to avoid ENOSPC*/
+		hashMax  uint32 = uint32(len(contents)) * 10
+		arrayMax uint32 = 4
+	)
+
+	hash := mustNewMap(Hash, hashMax)
 	defer hash.Close()
 
-	array, err := NewMap(&MapSpec{
-		Type:       Array,
-		KeySize:    4,
-		ValueSize:  4,
-		MaxEntries: 4,
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
+	array := mustNewMap(Array, arrayMax)
 	defer array.Close()
 
-	var keys, values []uint32
-	for key, value := range contents {
-		keys = append(keys, key)
-		values = append(values, value)
-	}
+	hashPerCpu := mustNewMap(PerCPUHash, hashMax)
+	defer hashPerCpu.Close()
 
-	for _, m := range []*Map{array, hash} {
+	arrayPerCpu := mustNewMap(PerCPUArray, arrayMax)
+	defer arrayPerCpu.Close()
+
+	for _, m := range []*Map{array, hash, arrayPerCpu, hashPerCpu} {
 		t.Run(m.Type().String(), func(t *testing.T) {
+			if m.Type() == PerCPUArray {
+				testutils.SkipOnOldKernel(t, "5.15", "BatchUpdate PerCPU Array")
```

Comment:
Can you add a link to the upstream commit where support was added? This can go into the commit message as well.

Reply:
```
				testutils.SkipOnOldKernel(t, "5.10", "BatchUpdate PerCPU Array")
				testutils.SkipOnOldKernel(t, "5.15", "BatchUpdate PerCPU Array")```
Ah I went by the versions were testing against, it's actually:

https://github.com/torvalds/linux/commit/f008d732ab181fd00d95c2e8a6e479d2f7c634b3 in 5.13.